### PR TITLE
(DOCSP-26980) Realm Reactify "Define a Realm Object Model" page

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

I didn't need to Bluehawk anything new here. Just used existing examples and adjusted surrounding text to fit the `@realm/react` context.

The page has IA/structure issues that we should address later on.

### Jira

- https://jira.mongodb.org/browse/DOCSP-26980

### Staged Changes

- [Define a Realm Object Model](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26980/sdk/react-native/realm-database/define-a-realm-object-model/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt [hunk 1]
```diff
@@ -45,81 +44,43 @@ If a realm already contains data when you open it, Realm Database
...
-          "properties": {
-            "isbn": "string",
-            "title": "string",
-            "author": "string",
-            "numberOwned": { "type": "int?", "default": 0 },
-            "numberLoaned": { "type": "int?", "default": 0 }
-          }
-        }
-      ]
+In ``@realm/react``, you define a realm schema by passing individual object
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/define-a-realm-object-model.txt:47]:
```suggestion
Using ``@realm/react``, you define a realm schema by passing individual object
```

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt [hunk 2]
```diff
@@ -45,81 +44,43 @@ If a realm already contains data when you open it, Realm Database
...
 
+.. literalinclude:: /examples/generated/react-native/js/RealmConfig.snippet.create-realm-context.js
+   :language: javascript
+   :emphasize-lines: 8
 
 .. _react-native-define-a-new-object-type:
 
-Define a New Object Type
+Define an Object Type
 ------------------------
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/define-a-realm-object-model.txt:57]:
```suggestion
Define an Object Type
---------------------
```

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt [hunk 3]
```diff
@@ -45,81 +44,43 @@ If a realm already contains data when you open it, Realm Database
...
+Define the type's ``name`` and ``properties`` in a static property called ``schema``. 
+The type's name must be unique among object types in a realm.
 
+.. tabs-realm-languages::
 
+   .. tab::
+      :tabid: typescript
 
-.. _react-native-define-objects-with-js-classes:
+      .. literalinclude:: /examples/generated/react-native/ts/Book.snippet.ts-book-schema.ts
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/define-a-realm-object-model.txt:68]: in the TS example, it says:

```ts
class Book extends Realm.Object<Book> {
```

i'm pretty sure you don't need `<Book>`. that looks odd to me if the class definition is self referential like this.

Comment (mongodben) [reply source/sdk/react-native/realm-database/define-a-realm-object-model.txt:68]: so i think this should be:

```ts
class Book extends Realm.Object {
```

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt [hunk 4]
```diff
@@ -45,81 +44,43 @@ If a realm already contains data when you open it, Realm Database
...
+      .. literalinclude:: /examples/generated/react-native/js/Book.snippet.js-book-schema.js
+        :language: javascript
 
-.. literalinclude:: /examples/generated/node/define-a-realm-object-schema.snippet.define-a-realm-object-schema-define-js-classes.js
-     :language: javascript
-   
-Pass the class itself to the schema property of the :js-sdk:`Realm.Configuration
+Then you can pass the class itself to the schema property of the :js-sdk:`Realm.Configuration
 <Realm.html#~Configuration>` object when opening a realm. You can then :doc:`read and write
 </sdk/node/examples/read-and-write-data>` normally.
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/define-a-realm-object-model.txt:79]:
```suggestion
<Realm.html#~Configuration>` object when opening a realm.
```

sentence doesn't really add value imo. also link is wrong and should be ref. also what's it mean to read/write data *normally*?

**Review by mongodben — APPROVED** [review]
lg&ast;tm in terms of @realm/reactification % few comments

&ast; as we discussed in slack/on the google doc, this page needs a full rewrite. but let's not get into that here.

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt (hunk 3 above)
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/define-a-realm-object-model.txt:68]: Ah, good point. It does seem unnecessary. I'm double checking on this. Many (but not all) of our TS models currently use the `extends Realm.Object<ObjectType>` approach.

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt (hunk 4 above)
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/define-a-realm-object-model.txt:79]: ha. Good points. This is from the current content. I debated whether to edit it or keep it as is. I opted to leave it so I didn't get tempted to rewrite the whole page. I'm glad you made this suggestion though! (I should've noticed the bad link. I fixed another of those on this page.)

File: source/sdk/react-native/realm-database/define-a-realm-object-model.txt (hunk 3 above)
Comment (takameyer) [reply source/sdk/react-native/realm-database/define-a-realm-object-model.txt:68]: It's odd, but it is currently correct.  This allows you to use the constructor to create a realm object and get a typed object back.

**Review by takameyer — APPROVED** [review]
LGTM

**Pull request merged**